src/components/LearnerActivityTable/data/hooks/useCourseEnrollments.js

```diff
@@ -0,0 +1,88 @@
+import {
+  useCallback, useMemo, useRef, useState,
+} from 'react';
+import { camelCaseObject } from '@edx/frontend-platform/utils';
+import debounce from 'lodash.debounce';
+import { logError } from '@edx/frontend-platform/logging';
+import EnterpriseDataApiService from '../../../../data/services/EnterpriseDataApiService';
+import EVENT_NAMES from '../../../../eventTracking';
+import { trackDataTableEvent } from '../utils';
+
+const applySortByToOptions = (sortBy, apiFieldsForColumnAccessor, options) => {
+  if (!sortBy || sortBy.length === 0) {
+    return;
+  }
+  const orderingStrings = sortBy.map(({ id, desc }) => {
+    const fieldForColumnAccessor = apiFieldsForColumnAccessor[id];
+    if (!fieldForColumnAccessor) {
+      return undefined;
+    }
+    const apiFieldKey = fieldForColumnAccessor.key;
+    return desc ? `-${apiFieldKey}` : apiFieldKey;
+  }).filter(orderingString => !!orderingString);
+  Object.assign(options, {
+    ordering: orderingStrings.join(','),
+  });
+};
+
+const useCourseEnrollments = (enterpriseId, tableId, apiFieldsForColumnAccessor) => {
+  const shouldTrackFetchEvents = useRef(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [courseEnrollments, setCourseEnrollments] = useState({
+    itemCount: 0,
+    pageCount: 0,
+    results: [],
+  });
+
+  const fetchCourseEnrollments = useCallback(async (args) => {
+    try {
+      setIsLoading(true);
+      const options = {
+        page: args.pageIndex + 1,
+        pageSize: args.pageSize,
+      };
+      applySortByToOptions(args.sortBy, apiFieldsForColumnAccessor, options);
+
+      const response = await EnterpriseDataApiService.fetchCourseEnrollments(enterpriseId, options);
+      const data = camelCaseObject(response.data);
+      setCourseEnrollments({
+        itemCount: data.count,
+        pageCount: data.numPages ?? Math.floor(data.count / options.pageSize),
+        results: data.results,
+      });
+
+      trackDataTableEvent({
+        shouldTrackRef: shouldTrackFetchEvents,
+        enterpriseId,
+        eventName: EVENT_NAMES.PROGRESS_REPORT.DATATABLE_SORT_BY_OR_FILTER,
+        tableId,
+        options,
+      });
+    } catch (error) {
+      logError(error);
+    } finally {
+      setIsLoading(false);
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
[nit] What dependencies are being left out that requires overriding the linter?

Reply:
`apiFieldsForColumnAccessor` is being used inside that hook, but we don't need to make it a dependency because it's going to be static, and when I added it to the dependency array, it created an unnecessary rendering issue.

Reply:
We should still be attempting to get to the root cause of the re-rendering issue and resolve it by memoizing the values.

Reply:
fixed.